src/main/cocoa/CocoaDisplay.mm

```diff
@@ -0,0 +1,390 @@
+            {
+                lsp_trace("handle = %p", handle);
+                CocoaWindow *wnd = new CocoaWindow(this, NULL, false);
+                add_window(wnd);
+                return wnd;
+            }
+
+            bool CocoaDisplay::add_window(CocoaWindow *wnd)
+            {
+                return vWindows.add(wnd);
+            }
+
+            bool CocoaDisplay::remove_window(CocoaWindow *wnd)
+            {
+                // Remove focus window
+                //if (pFocusWindow == wnd)
+                //    pFocusWindow = NULL;
+
+                // Remove window from list
+                if (!vWindows.premove(wnd))
+                    return false;
+
+                // Check if need to leave main cycle
+                if (vWindows.size() <= 0)
+                    bExit = true;
+                return true;
+            }
+
+            CocoaWindow *CocoaDisplay::find_window(void *wnd) {
+                NSWindow *nswnd = (__bridge NSWindow *)wnd;
+                
+                size_t n = vWindows.size();
+
+                for (size_t i = 0; i < n; ++i) {
+                    CocoaWindow *w = vWindows.uget(i);
+                    if (w == NULL)
+                        continue;
+                    if (w->nswindow() == nswnd)
+                        return w;
+                }
+
+                return NULL;
+            }
+
+
+            void CocoaDisplay::destroy()
+            {
+                {
+                    sTasksLock.lock();
+                    lsp_finally { sTasksLock.unlock(); };
+                }
+                IDisplay::destroy();
+            }
+
+            const MonitorInfo *CocoaDisplay::enum_monitors(size_t *count) {
+                // Prepare result array
+                lltl::darray<MonitorInfo> result;
+
+                NSArray<NSScreen *> *screens = [NSScreen screens];
+                NSUInteger nmonitors = [screens count];
+
+                MonitorInfo *items = result.add_n(nmonitors);
+                if (items == nullptr)
+                    return nullptr;
+
+                for (NSUInteger i = 0; i < nmonitors; ++i) {
+                    MonitorInfo *di = &items[i];
+                    new (&di->name, inplace_new_tag_t()) LSPString;
+
+                    NSScreen *screen = screens[i];
+                    NSRect frame = [screen frame];
+
+                    // Set screen bounds
+                    di->rect.nLeft   = static_cast<int>(frame.origin.x);
+                    di->rect.nTop    = static_cast<int>(frame.origin.y);
+                    di->rect.nWidth  = static_cast<int>(frame.size.width);
+                    di->rect.nHeight = static_cast<int>(frame.size.height);
+
+                    // Set primary flag (main screen)
+                    di->primary = (screen == [NSScreen mainScreen]);
+
+                    // No native way to get monitor name in Cocoa, so use fallback
+                    char buf[32];
```

Comment:
We can use `fmt_ascii` or `fmt_utf8` method of `LSPString` to not to use temporary buffers on stack.